7/Dockerfile.rhel7

```diff
@@ -1,51 +1,50 @@
 
-ENV SUMMARY="Red Hat LLVM Toolset 7 container image" \
-    DESCRIPTION="Platform for building applications using Red Hat LLVM Toolset 7"
+EXPOSE 8080
 
-LABEL com.redhat.component="llvm-toolset-7-docker" \
-      name="devtools/llvm-toolset-7-rhel7" \
-      version="7" \
-      summary="$SUMMARY" \
+ENV LLVM_VERSION="4.0.1" \
+    SUMMARY="Platform for building projects with clang/llvm" \
+    DESCRIPTION="$SUMMARY"
+
+
+LABEL summary="$SUMMARY" \
       description="$DESCRIPTION" \
       io.k8s.description="$DESCRIPTION" \
-      io.k8s.display-name="LLVM Toolset 7"
+      io.k8s.display-name="LLVM 4.0.1" \
+      io.openshift.expose-services="8080:http" \
+      io.openshift.tags="builder,llvm,clang" \
+      com.redhat.component="llvm-toolset-7-docker" \
+      name="devtools/llvm-toolset-7-rhel7" \
+      version="1" \
+      maintainer="SoftwareCollections.org <sclorg@redhat.com>"
 
-# To use subscription inside container yum command has to be run first (before yum-config-manager)
-# https://access.redhat.com/solutions/1443553
-RUN yum repolist > /dev/null && \
-    yum-config-manager --disable \* &> /dev/null && \
-    yum-config-manager --enable rhel-server-rhscl-7-rpms && \
-    yum-config-manager --enable rhel-server-rhscl-7-beta-rpms && \
+RUN yum install -y centos-release-scl-rh && \
     yum-config-manager --enable rhel-7-server-devtools-rpms && \
-    yum-config-manager --enable rhel-7-server-rpms && \
+    yum-config-manager --enable rhel-server-rhscl-7-rpms && \
     yum-config-manager --enable rhel-7-server-optional-rpms && \
-    INSTALL_PKGS="llvm-toolset-7" && \
+    yum-config-manager --disable epel >/dev/null || : && \
+    INSTALL_PKGS="llvm-toolset-7 llvm-toolset-7-cmake" && \
     yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
     rpm -V $INSTALL_PKGS && \
     yum clean all -y
 
-COPY ./root /
-
-ENV HOME=/opt/app-root/src \
-    PATH=/opt/app-root/src/bin:/opt/app-root/bin:/opt/rh/llvm-toolset-7/root/usr/bin/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+COPY ./s2i/bin/ $STI_SCRIPTS_PATH
 
-RUN mkdir -p ${HOME} && \
-    groupadd -r default -f -g 1001 && \
-    useradd -u 1001 -r -g default -d ${HOME} -s /sbin/nologin \
-    -c "Default Application User" default && \
-    chown -R 1001:1001 /opt/app-root /usr/bin/container-entrypoint /usr/bin/usage && \
-    chmod u+x /usr/bin/usage
-
-USER 1001
-
-WORKDIR ${HOME}
+# Drop the root user and make the content of /opt/app-root owned by user 1001
+RUN chown -R 1001:0 /opt/app-root && \
+    chmod -R ug+rwx /opt/app-root && \
+    rpm-file-permissions
 
 # Enable the SCL for all bash scripts.
 ENV BASH_ENV=/opt/app-root/etc/scl_enable \
     ENV=/opt/app-root/etc/scl_enable \
-    PROMPT_COMMAND=". /opt/app-root/etc/scl_enable"
+    PROMPT_COMMAND=". /opt/app-root/etc/scl_enable" \
+    PATH=/opt/rh/llvm-toolset-7/root/usr/bin:$PATH \
+    CC="clang" \
+    CXX="clang++"
+
+USER 1001
+
+RUN source scl_source enable llvm-toolset-7 || true
```

Comment:
I'm afraid this is not working as expected -- if it was supposed to enable the collection, then I don't think it does the trick. I might be wrong though. What we use for enabling the collection is the trick with sourcing `/opt/app-root/etc/scl_enable` file by setting `BASH_ENV`, `PROMPT_COMMAND`, and `ENV`, which is already done in s2i-base image: https://github.com/sclorg/s2i-base-container/blob/master/core/Dockerfile#L35

Reply:
Ah, I see these variables are defined in this Dockerfile as well, which should not make it worse, but it's probably not necessary.